pkg/objectio/cache.go

```diff
@@ -47,7 +47,14 @@ func LoadObjectMetaByExtent(
 	fs fileservice.FileService) (meta ObjectMeta, err error) {
 	v, _, ok := metaCache.Get(*name.Short(), false)
 	if ok {
-		meta = v.(ObjectMeta)
+
```

Comment:
sigh. how could decode/unmarshal be any faster than type assertion? 
and casting []byte to other types is not safer than type assertion.
this will hurt the performance.

Reply:
1. For this part alone, ie Decode(v, `true`), there is **No Unmarshalling**. 

```go
func Decode(buf []byte, noHeaderHint bool) (any, error) {
	if noHeaderHint {
		return buf, nil
	}

	header := DecodeIOEntryHeader(buf)
	codec := GetIOEntryCodec(*header)
	if codec.NoUnmarshal() {
		return buf[IOEntryHeaderSize:], nil
	}
	v, err := codec.Decode(buf[IOEntryHeaderSize:])
	if err != nil {
		return nil, err
	}
	return v, nil
}

```
I still use `Decode()` here to preserve the old `ToObject` functionality.

2. You are right. For other parts where we are doing Decode(v, `false`), we might have a performance impact due to frequent Unmarshalling/Decoding of `byte slice` returned from MemCache.

Reply:
Why do you need the obj any here?    All you need is to decode into meta, no?  Here the right thing is a pointer cast.

Reply:
We can remove it later. As mentioned in the PR description, this PR is not yet completed.

I will remove `any`, after successfully validating the BVT test case.

Reply:
Removed obj any here.